model/simulationtests/generator_windturbine.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require 'openstudio'
+require 'lib/baseline_model'
+
+model = BaselineModel.new
+
+# make a 1 story, 100m X 50m, 5 zone core/perimeter building
+model.add_geometry({ 'length' => 100,
+                     'width' => 50,
+                     'num_floors' => 1,
+                     'floor_to_floor_height' => 4,
+                     'plenum_height' => 1,
+                     'perimeter_zone_depth' => 3 })
+
+# add windows at a 40% window-to-wall ratio
+model.add_windows({ 'wwr' => 0.4,
+                    'offset' => 1,
+                    'application_type' => 'Above Floor' })
+
+# add ASHRAE System type 01, PTAC, Residential
+model.add_hvac({ 'ashrae_sys_num' => '01' })
+
+# add thermostats
+model.add_thermostats({ 'heating_setpoint' => 24,
+                        'cooling_setpoint' => 28 })
+
+# assign constructions from a local library to the walls/windows/etc. in the model
+model.set_constructions
+
+# set whole building space type; simplified 90.1-2004 Large Office Whole Building
+model.set_space_type
+
+# add design days to the model (Chicago)
+model.add_design_days
+
+# add wind turbine generators
+generator1 = OpenStudio::Model::GeneratorWindTurbine.new(model)
+
+generator2 = OpenStudio::Model::GeneratorWindTurbine.new(model)
+generator2.setRotorType('VerticalAxisWindTurbine')
+generator2.setPowerControl('FixedSpeedFixedPitch')
+generator2.setBladeChordArea(100.0)
+
+generator3 = OpenStudio::Model::GeneratorWindTurbine.new(model)
+generator3.setPowerControl('FixedSpeedVariablePitch')
+
+generator4 = OpenStudio::Model::GeneratorWindTurbine.new(model)
+generator4.setPowerControl('VariableSpeedFixedPitch')
```

Comment:
In the interest of demonstrating the API, I would create Two GeneratorWindTurbine, one Vertical and one Horizontal, and set all related (required) fields for that specific configuration + all common fields. So that every setter is called.

Actually, let's make it three, use the ones ffrom the E+ example file.

Reply:
```
s1 = """  Generator:WindTurbine,
    WT1,                     !- Name
    Always On,               !- Availability Schedule Name
    HorizontalAxisWindTurbine,  !- Rotor Type
    FixedSpeedVariablePitch, !- Power Control
    41,                      !- Rated Rotor Speed {rev/min}
    19.2,                    !- Rotor Diameter {m}
    30.5,                    !- Overall Height {m}
    3,                       !- Number of Blades
    55000,                   !- Rated Power {W}
    11,                      !- Rated Wind Speed {m/s}
    3.5,                     !- Cut In Wind Speed {m/s}
    25,                      !- Cut Out Wind Speed {m/s}
    0.835,                   !- Fraction system Efficiency
    8,                       !- Maximum Tip Speed Ratio
    0.5,                     !- Maximum Power Coefficient
    6.4,                     !- Annual Local Average Wind Speed {m/s}
    50,                      !- Height for Local Average Wind Speed {m}
    ,                        !- Blade Chord Area {m2}
    ,                        !- Blade Drag Coefficient
    ,                        !- Blade Lift Coefficient
    0.5176,                  !- Power Coefficient C1
    116,                     !- Power Coefficient C2
    0.4,                     !- Power Coefficient C3
    0,                       !- Power Coefficient C4
    5,                       !- Power Coefficient C5
    21;                      !- Power Coefficient C6"""

s2 = """Generator:WindTurbine,
    WT2,                     !- Name
    Always On,               !- Availability Schedule Name
    HorizontalAxisWindTurbine,  !- Rotor Type
    FixedSpeedFixedPitch,    !- Power Control
    59,                      !- Rated Rotor Speed {rev/min}
    21,                      !- Rotor Diameter {m}
    37,                      !- Overall Height {m}
    3,                       !- Number of Blades
    100000,                  !- Rated Power {W}
    14.5,                    !- Rated Wind Speed {m/s}
    3.5,                     !- Cut In Wind Speed {m/s}
    25,                      !- Cut Out Wind Speed {m/s}
    0.835,                   !- Fraction system Efficiency
    7,                       !- Maximum Tip Speed Ratio
    0.23,                    !- Maximum Power Coefficient
    6.4,                     !- Annual Local Average Wind Speed {m/s}
    50,                      !- Height for Local Average Wind Speed {m}
    ,                        !- Blade Chord Area {m2}
    ,                        !- Blade Drag Coefficient
    ,                        !- Blade Lift Coefficient
    ,                        !- Power Coefficient C1
    ,                        !- Power Coefficient C2
    ,                        !- Power Coefficient C3
    ,                        !- Power Coefficient C4
    ,                        !- Power Coefficient C5
    ;                        !- Power Coefficient C6"""

s3 = """Generator:WindTurbine,
    WT3,                     !- Name
    Always On,               !- Availability Schedule Name
    VerticalAxisWindTurbine, !- Rotor Type
    VariableSpeedFixedPitch, !- Power Control
    130,                     !- Rated Rotor Speed {rev/min}
    5.2,                     !- Rotor Diameter {m}
    11,                      !- Overall Height {m}
    3,                       !- Number of Blades
    10000,                   !- Rated Power {W}
    11,                      !- Rated Wind Speed {m/s}
    3,                       !- Cut In Wind Speed {m/s}
    25,                      !- Cut Out Wind Speed {m/s}
    0.75,                    !- Fraction system Efficiency
    5,                       !- Maximum Tip Speed Ratio
    ,                        !- Maximum Power Coefficient
    6.4,                     !- Annual Local Average Wind Speed {m/s}
    50,                      !- Height for Local Average Wind Speed {m}
    2.08,                    !- Blade Chord Area {m2}
    0.9,                     !- Blade Drag Coefficient
    0.05,                    !- Blade Lift Coefficient
    ,                        !- Power Coefficient C1
    ,                        !- Power Coefficient C2
    ,                        !- Power Coefficient C3
    ,                        !- Power Coefficient C4
    ,                        !- Power Coefficient C5
    ;                        !- Power Coefficient C6"""

for k, s in enumerate([s1, s2, s3]):
    for i, line in enumerate(s.splitlines()):
        if i == 0:
            continue
        val, field = line.split('!-')
        field = field.split('{')[0].strip().replace(' ', '')
        val = val.replace(',','').replace(';', '').strip()
        is_numeric = False
        try:
            val = float(val)
            is_numeric = True
        except:
            if val == "Always On":
                val = "model.alwaysOnDiscreteSchedule()"
            else:
                val = f'"{val}"'
        if val == '""':
            print(f"// generator{k+1}.set{field}( )")
        else:
            print(f"generator{k+1}.set{field}({val})")
        
    print('\n\n')
```

Reply:
```
generator1.setName("WT1")
generator1.setAvailabilityScheduleName(model.alwaysOnDiscreteSchedule())
generator1.setRotorType("HorizontalAxisWindTurbine")
generator1.setPowerControl("FixedSpeedVariablePitch")
generator1.setRatedRotorSpeed(41.0)
generator1.setRotorDiameter(19.2)
generator1.setOverallHeight(30.5)
generator1.setNumberofBlades(3.0)
generator1.setRatedPower(55000.0)
generator1.setRatedWindSpeed(11.0)
generator1.setCutInWindSpeed(3.5)
generator1.setCutOutWindSpeed(25.0)
generator1.setFractionsystemEfficiency(0.835)
generator1.setMaximumTipSpeedRatio(8.0)
generator1.setMaximumPowerCoefficient(0.5)
generator1.setAnnualLocalAverageWindSpeed(6.4)
generator1.setHeightforLocalAverageWindSpeed(50.0)
// generator1.setBladeChordArea( )
// generator1.setBladeDragCoefficient( )
// generator1.setBladeLiftCoefficient( )
generator1.setPowerCoefficientC1(0.5176)
generator1.setPowerCoefficientC2(116.0)
generator1.setPowerCoefficientC3(0.4)
generator1.setPowerCoefficientC4(0.0)
generator1.setPowerCoefficientC5(5.0)
generator1.setPowerCoefficientC6(21.0)



generator2.setName("WT2")
generator2.setAvailabilityScheduleName(model.alwaysOnDiscreteSchedule())
generator2.setRotorType("HorizontalAxisWindTurbine")
generator2.setPowerControl("FixedSpeedFixedPitch")
generator2.setRatedRotorSpeed(59.0)
generator2.setRotorDiameter(21.0)
generator2.setOverallHeight(37.0)
generator2.setNumberofBlades(3.0)
generator2.setRatedPower(100000.0)
generator2.setRatedWindSpeed(14.5)
generator2.setCutInWindSpeed(3.5)
generator2.setCutOutWindSpeed(25.0)
generator2.setFractionsystemEfficiency(0.835)
generator2.setMaximumTipSpeedRatio(7.0)
generator2.setMaximumPowerCoefficient(0.23)
generator2.setAnnualLocalAverageWindSpeed(6.4)
generator2.setHeightforLocalAverageWindSpeed(50.0)
// generator2.setBladeChordArea( )
// generator2.setBladeDragCoefficient( )
// generator2.setBladeLiftCoefficient( )
// generator2.setPowerCoefficientC1( )
// generator2.setPowerCoefficientC2( )
// generator2.setPowerCoefficientC3( )
// generator2.setPowerCoefficientC4( )
// generator2.setPowerCoefficientC5( )
// generator2.setPowerCoefficientC6( )



generator3.setName("WT3")
generator3.setAvailabilityScheduleName(model.alwaysOnDiscreteSchedule())
generator3.setRotorType("VerticalAxisWindTurbine")
generator3.setPowerControl("VariableSpeedFixedPitch")
generator3.setRatedRotorSpeed(130.0)
generator3.setRotorDiameter(5.2)
generator3.setOverallHeight(11.0)
generator3.setNumberofBlades(3.0)
generator3.setRatedPower(10000.0)
generator3.setRatedWindSpeed(11.0)
generator3.setCutInWindSpeed(3.0)
generator3.setCutOutWindSpeed(25.0)
generator3.setFractionsystemEfficiency(0.75)
generator3.setMaximumTipSpeedRatio(5.0)
// generator3.setMaximumPowerCoefficient( )
generator3.setAnnualLocalAverageWindSpeed(6.4)
generator3.setHeightforLocalAverageWindSpeed(50.0)
generator3.setBladeChordArea(2.08)
generator3.setBladeDragCoefficient(0.9)
generator3.setBladeLiftCoefficient(0.05)
// generator3.setPowerCoefficientC1( )
// generator3.setPowerCoefficientC2( )
// generator3.setPowerCoefficientC3( )
// generator3.setPowerCoefficientC4( )
// generator3.setPowerCoefficientC5( )
// generator3.setPowerCoefficientC6( )

```